javaparser-core/pom.xml

```diff
@@ -79,7 +79,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>1.4.1</version>
+                <version>3.0.0-M1</version>
```

Comment:
This M1 should scare us?

Reply:
Not as much as 1.4.2-jenkins1 ;-) After some hunting for release notes I found this: http://maven.40175.n5.nabble.com/ANN-Apache-Maven-Enforcer-Plugin-3-0-0-M1-Released-td5911744.html

Reply:
Anyway, the alternatives are worse in my opinion: either take out the JDK 8 API enforcement for core, or don't compile with JDK9 at all.